packages/smithy-http/src/smithy_http/aio/crt.py

```diff
@@ -251,7 +265,11 @@ async def send(
         crt_stream.completion_future.add_done_callback(
             partial(self._close_input_body, body=crt_body)
         )
-        return await response_factory.await_response()
+        try:
+            return await response_factory.await_response()
+        except CRTErrorResponse as e:
+            await close(crt_body)
+            return e.response
```

Comment:
I'm not a fan of using exceptions for control flow like this. (I know python does it in the standard library but I don't like that either.) Why not just do this:

```suggestion
        response = await response_factory.await_response()
        if response.status != 200 and response.status >= 300:
            await close(crt_body)
        return response
```

Reply:
Yeah - good thought - I like checking the response status here better.